File: backend/src/main/java/com/navalgo/backend/fleet/Owner.java
package com.navalgo.backend.fleet;

import com.navalgo.backend.company.Company;
import jakarta.persistence.*;

@Entity
@Table(name = "owners")
public class Owner {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OwnerType type;

    @Column(nullable = false)
    private String displayName;

    @Column(nullable = false)
    private String documentId;

    private String phone;
    private String email;

    @ManyToOne(fetch = FetchType.LAZY)
    private Company company;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public OwnerType getType() { return type; }
    public void setType(OwnerType type) { this.type = type; }

    public String getDisplayName() { return displayName; }
    public void setDisplayName(String displayName) { this.displayName = displayName; }

    public String getDocumentId() { return documentId; }
    public void setDocumentId(String documentId) { this.documentId = documentId; }

    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public Company getCompany() { return company; }
    public void setCompany(Company company) { this.company = company; }
}

File: lib/models/owner.dart
class Owner {
  const Owner({
    required this.id,
    required this.type,
    required this.displayName,
    required this.documentId,
    this.phone,
    this.email,
    this.companyId,
  });

  final int id;
  final String type;
  final String displayName;
  final String documentId;
  final String? phone;
  final String? email;
  final int? companyId;

  factory Owner.fromJson(Map<String, dynamic> json) {
    return Owner(
      id: json['id'] as int,
      type: json['type'] as String,
      displayName: json['displayName'] as String,
      documentId: json['documentId'] as String,
      phone: json['phone'] as String?,
      email: json['email'] as String?,
      companyId: json['companyId'] as int?,
    );
  }
}
